core/src/main/java/io/questdb/griffin/WhereClauseParser.java

```diff
@@ -645,142 +661,148 @@ private boolean analyzeInInterval(
         if (!isTimestamp(col)) {
             return false;
         }
-
-        if (in.paramCount == 2) {
-            ExpressionNode inArg = in.rhs;
-            if (inArg.type == ExpressionNode.CONSTANT) {
-                // Single value ts in '2010-01-01' - treat string literal as an interval, not single Timestamp point
-                if (isNullKeyword(inArg.token)) {
-                    if (!isNegated) {
-                        model.intersectIntervals(Numbers.LONG_NULL, Numbers.LONG_NULL);
-                    } else {
-                        model.subtractIntervals(Numbers.LONG_NULL, Numbers.LONG_NULL);
-                    }
-                } else {
-                    if (!isNegated) {
-                        model.intersectIntervals(inArg.token, 1, inArg.token.length() - 1, inArg.position);
+        int oldIntervalFuncType = executionContext.getIntervalFunctionType();
+        try {
+            executionContext.setIntervalFunctionType(ColumnType.getIntervalType(timestampDriver.getColumnType()));
```

Comment:
`IntrinsicModel` is where the state of the parsing is. Using execution context is not a good idea, execution context is a broader entity. Also, what would happen if we mix function types, e.g. micro- and nano- ?

Reply:
Yes, `IntervalFunctionType` is for dynamic interval functions like `today()` and `tomorrow()`. It determines the required precision of the interval (micros or nanos) through the surrounding context.

Alternatively, if this complexity isn't needed, these dynamic functions can use the highest time precision: nanos.It just like this way, we can control things more precisely.

Outside of a `WHERE` clause, `IntervalFunctionType` uses the default value from `SqlExecutionContextImpl`, which is `Micros`.